src/contexts/BuyContext.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useState, createContext, useMemo } from 'react';
+
+export function BuyProvider({ children }) {
+  const [cartData, setCartData] = useState([]);
+
+  const addToCart = (newTextbook) => {
+    setCartData((currentCartData) => {
+      // Check if the item already exists in the cart
+      const itemExists = currentCartData.some((item) => item.id === newTextbook.id);
+
+      // If the item does not exist, add it to the cart
+      if (!itemExists) {
+        return [newTextbook, ...currentCartData];
```

Comment:
hmm dont wouldnt we want to` setCartData`

Reply:
this is within the setCartData function